plugins/thunderbolt/fu-self-test.c

```diff
@@ -792,18 +793,21 @@ static MockDevice root_one = {
 	.name = "Laptop",
 	.id = "0x23",
 	.nvm_version = "20.0",
+	.nvm_parsed_version = "20.00",
```

Comment:
With this change, I wonder if it's also adding a device that changes in potentially non-obvious way to self tests since it's not obvious immediately that's what's happening.
eg `.nvm_version = 2.2` -> `.nvm_parsed_version = 02.02`

Reply:
Not sure I follow. `nvm_parsed_version` is the expected version string after fwupd has added the corresponding device. It uses the format `%02X.%02X` so 2.2 would be 02.02. But that's exactly what happens with real devices, no?

Reply:
Yes, but it's not an obvious behavior.  At least when you're given a string `X.Y` but `X` and `Y` are one digit usually you treat it as `0X.0Y`.  So when the interface from the kernel gives `23.5` unless you've been watching closely you won't know that's major 23 and minor 5 and might read it as `23.50`.

I just want to make sure that nuance is captured in the self tests as the people who aren't us who look at this code down the road might not realize that's what happening.

So I think either adding another device to self tests or changing one of them to have a non-zero minor would be good.

Reply:
OK, understood now.